pkg/client/sharkey_client.go

```diff
@@ -206,6 +206,10 @@ func (c *Client) makeKnownHosts() {
 }
 
 func (c *Client) GenerateClient() error {
+	if c.client != nil {
```

Comment:
The `GenerateClient()` bit seems suboptimal, given that we probably only care that we reload the keypair occasionally. There probably better way to solve it, like setting a callback on `tls#Config` for `GetClientCertificate` or something like that. Until we decide to make a bigger refactoring, this seems fine, though.